.teamcity/_self/projects/WPComPlugins.kt

```diff
@@ -184,3 +188,54 @@ private object InlineHelp : WPComPluginBuild(
 	docsLink = "TODO",
 	withPRNotify = "false",
 )
+
+private object GutenbergUploadSourceMapsToSentry: BuildType() {
+	init {
+		name = "Upload Gutenberg Source Maps to Sentry";
+		id("WPComPlugins_GutenbergUploadSourceMapsToSentry");
+
+		// vcs {
+		//	GitVcsRoot {
+		//		name = "gutenberg"
+		//		url = "git@github.com:fullofcaffeine/gutenberg.git"
+		//		pushUrl = "git@github.com:fullofcaffeine/gutenberg.git"
+		//		branch = "refs/heads/trunk"
+		//		branchSpec = "+:refs/heads/*"
+		//		useTagsAsBranches = true
+		//		authMethod = uploadedKey {
+		//			uploadedKey = "matticbot"
+		//		}
+		//	}
+		// }
+
+		params {
+			text(
+				name = "GUTENBERG_VERSION",
+				value = "",
+				label = "Gutenberg version",
+				description = "The Gutenberg version to upload source maps for (include the whole string, including the `v` prefix)",
+				allowEmpty = false
+			)
+		}
+
+		steps {
+			mergeTrunk()
+			bashNodeScript {
+				name = "Upload source maps to Sentry"
+				scriptContent = """
+					# Install the Sentry CLI binary
+					curl -sL https://sentry.io/get-cli/ | bash
+
+					wget https://github.com/WordPress/gutenberg/releases/tag/%GUTENBERG_VERSION%/gutenberg.zip
+					unzip gutenberg.zip -d gutenberg
+					cd gutenberg
+
+					sentry-cli --auth-token %SENTRY_AUTH_TOKEN% releases --org a8c files "wpcom-test-01" delete --all
```

Comment:
I'm guessing we'll need to pass the release via a paramater?

So we'd basically create some kind of release identifier on wpcom, update it in the sentry ETK code (?) and then also make sure it's used here?

Part of the challenge will be keeping the version updated on wpcom without too much overhead.

Reply:
Yes, exactly. The release is being passed there as "wpcom-test-01". I think it's fine to leave it as "wpcom-test-01" for now, but I tried to elaborate on this problem in the PR description, quoting here: 

```
The ideal implementation will use a new Sentry release for each deployment. The problem is that Sentry is not exclusive to Gutenberg, and its instance shared across the WPAdmin context, which means that a release for the WPAdmin project can't have metadata exclusive to a certain Gutenberg release. We can't create a new Sentry project for Gutenberg either, as it needs to be specified in the Sentry configuration in the frontend, and it's shared across the execution spae for GB, ETK, et al.

The best we can do, I think, is to just have sequential releases, like each release of Gutenberg (which will trigger this build) and other WPCOM plugins just create a new sequential release on Sentry.

Another possibility is to look at the WPCOM git repo as the consolidator of all these apps that are tracked by the Sentry WPadmin project in the frontend. We can somehow get the commit hash for a deployment and use it to create the new release in Sentry. This has another race-condition downside though: we can only get the hash (AFAIK) after the deployment is done, so in theory, there's going to be a window of time where the release will not have source-maps. Also, we should remember that for each of those releases, the hash should be committed together so that the Sentry configuration in error-reporting [can access it and pass it to Sentry](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/error-reporting/index.js#L21) (i.e window.WPCOM_RELEASE or something like that)
```